### ISSUE TITLE
Compilation failure with C++11

### ISSUE THREAD

**Issue by galexv** [ALPSCore/ALPSCore#401]
Compilation with C++11 complains about ambiguous overloads when a dictionary element is assigned, such as:
```c++
typedef std::vector<int> vector;
vector v;
// Overload in question:
v=dict["name"];
// Candidates:
vector::operator=(const vector& );
vector::operator=(vector&& );
vector::operator=(std::initializer_list<int> );
```

**Comment by galexv** [comment]
The problem is, of course, that the dictionary value type is "convertible to anything".
I have limited convertibility to "supported types" only, via SFINAE (in a manner that does not work in C++03, so had to add a conditional compilation around it).
I'd like to find a more elegant solution.

**Comment by egull** [comment]
OK, so this seems fixed but I do get a new test failure:

[ RUN      ] DictionaryTest0.storeStringGetCharPtr
/Users/egull/ALPSCore/params/test/dictionary.cpp:196: Failure
Value of: std::string(actual)
  Actual: ""
Expected: expected
Which is: "Hello, world!"
[  FAILED  ] DictionaryTest0.storeStringGetCharPtr (0 ms)
[----------] 14 tests from DictionaryTest0 (1 ms total)

**Comment by galexv** [comment]
Cannot reproduce it on linux wiith clang 3.8.0
@egull: Could you please send me your `CMakeCache.txt`? Thanks!

**Comment by galexv** [comment]
Tried on Linux with clang 4.0.0 (and `-std=c++11`), all tests pass as well.

**Comment by galexv** [comment]
Fix issued in commit 33c5c1a

**Comment by galexv** [comment]
@egull I have merged PR #403 (issue #402); the fix for https://github.com/ALPSCore/ALPSCore/issues/401#issuecomment-348624851 is already in the code (commit https://github.com/ALPSCore/ALPSCore/commit/33c5c1a336d5d73201b91fa2c2e030642737c269).

**Comment by egull** [comment]
OK. I can confirm that the comment in 401 fixes the test failure.
